packages/contracts/contracts/CollateralManagerV2.sol

```diff
@@ -0,0 +1,496 @@
+pragma solidity >=0.8.0 <0.9.0;
+// SPDX-License-Identifier: MIT
+
+/*
+
+1. During submitBid, the collateral will be Committed (?) using the 'collateral validator'
+
+2. During acceptBid, the collateral gets bundled into the CollateralBundler which mints an NFT (the bundle) which then gets transferred into this contract 
+
+
+This collateral manager will only accept collateral bundles. 
+
+*/
+
+// Contracts
+import "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+
+// Libraries
+import "@openzeppelin/contracts-upgradeable/utils/structs/EnumerableSetUpgradeable.sol";
+
+// Interfaces
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC1155/IERC1155Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC721/IERC721ReceiverUpgradeable.sol";
+import "./interfaces/ICollateralManagerV2.sol";
+//import { Collateral, CollateralType, ICollateralEscrowV1 } from "./interfaces/escrow/ICollateralEscrowV1.sol";
+import "./interfaces/ITellerV2.sol";
+import "./bundle/TokenStore.sol";
+
+import "./bundle/interfaces/ICollateralBundle.sol";
+
+//remove me 
+import "lib/forge-std/src/console.sol";
+
+/*
+
+This contract is a token store which stores bundles.
+The bid id == the bundle id. 
+
+If the bundle exists and is owned by this contract, we know the collateral is held. 
+
+*/
+
+contract CollateralManagerV2 is
+    ContextUpgradeable,
+    TokenStore,
+    ICollateralManagerV2
+{
+    /* Storage */
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+    ITellerV2 public tellerV2;
+
+    // bidIds -> collateralEscrow
```

Comment:
research: 

Add a mapping that maps BidID to BundleID 

should make rollover easy as we can , in this level of the code, remap that